### PR TITLE
[mlir][vector] Update reduction kind docs. NFC.

### PR DESCRIPTION
**Pull request by kuhar**
Update the documentation surrounding reduction kinds. Highlight different min/max reduction kinds for signed/unsigned integers and floats. Update IR examples.

### CODE REVIEW

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td [hunk 1]
```diff
@@ -937,11 +940,12 @@ def Vector_OuterProductOp :
...
-    for int/fp, and and/or/xor for int only. The default is "add", in which
-    case the operation returns a fused multiply-add. In other cases it returns
-    a multiply followed by the appropriate operation (for example, a compare and
-    select for "max").
+    An optional kind attribute may be specified to be: `add`/`mul`/`minsi`
+    /`minui`/`maxsi`/`maxui`/`and`/`or`/`xor` for integers, and `add`/`mul`
+    /`minf`/`maxf`/`minimumf`/`maximumf` for floats.
+    The default is `add`, in which case the operation returns a fused
+    multiply-add. In other cases it returns a multiply followed by the
+    appropriate operation (for example, a compare and select for `maxf`).
```
Comment (dcaballe) [inline mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:948]: I think the "compare and select" is not longer accurate?

**Review by dcaballe — APPROVED** [review]
Thanks! Just one comment

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td (hunk 1 above)
Comment (kuhar) [reply mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:948]: what should it be instead?  I'm not familiar with this one

**Review by bjacob — APPROVED** [review]

File: mlir/include/mlir/Dialect/Vector/IR/VectorOps.td (hunk 1 above)
Comment (kuhar) [reply mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:948]: This can be a follow up fix -- just let me know what wording you'd like to see here

Comment (dcaballe) [reply mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:948]: I would remove everything after "The default is `add`. "The operation returns a fused multiply-add" seems to be talking about the lowering. Same for the `maxf`. We generate one of the max operations in the arith dialect. THis is probably something that was written quite some time ago.

Comment (kuhar) [reply mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:948]: https://github.com/llvm/llvm-project/pull/70824

**Pull request merged**